### PR TITLE
Enable SSR for migration banner

### LINKED ISSUE #515: CC Migration banner doesn't SSR
## Description

The migration banner does not render server side so it "pops in" after the page is hydrated on the client.

## Reproduction

1. Visit https://wordpress.org/openverse/?referrer=creativecommons.org
2. Watch the top of the Openverse iframe and note the banner popping in shortly after the page finishes loading

## Expectation
The banner should be SSR rendered. We have the query param already (and we could check the referrer header anyway) so it's definitely possible to know that it should be present server side.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #515 by @sarayourfriend

## Description

This PR enables the migration banner to render via SSR.

## Cause
Nuxt middleware runs once on the server during the first render and subsequent times on the client during routing. On the server, the plugin `migration-notice.js` sets the Vuex state `isReferredByCc` to true, and the middleware immediately sets it to false. So the banner isn't part of the SSR page.

On the client the plugin again sets the Vuex state `isReferredByCc` to true. The middleware already having executed on the server is not run on the client for the first page, so the banner pops in. Further navigation causes the middleware to run and remove the banner (as expected).

By changing the middleware to not run the unset mutation on the server, we are able to ensure that it only has the intended banner clearing effect on the client.

This problem was made harder to debug because the layout was non-reactively using the store state.

## Testing Instructions

0. Check out the PR and run the dev server.
1. Open the URL http://localhost:8443/search/image?q=cat&referrer=creativecommons.org.
2. Observe that the migration banner is SSR-ed.
3. Navigate to any different page.
4. Observe that the banner goes away and never returns.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/middleware/middleware.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ export default function ({ store, query, route }) {
     })
   }
 
-  if (store.state.nav.isReferredFromCc) {
+  if (!process.server && store.state.nav.isReferredFromCc) {
```
Comment (sarayourfriend) [inline src/middleware/middleware.js:31]: nit: FWIW positive assertions always seem easier for me to quickly grok than a negation. Clearer for new folks or someone skimming as well who might not have the full context of `process.server` and `process.client`.

```suggestion
  if (process.client && store.state.nav.isReferredFromCc) {
```

**Review by sarayourfriend — APPROVED** [review]
LGTM! Nice and easy fix :tada: 

Can you add an E2E test for this? :thinking:

**Comment by obulat** [comment]
If I understand correctly, the migration banner can only be displayed on the homepage, because that's the only page that CC Search redirects to. So, we can remove the migration banner from the default layout. In fact, we could even move the MigrationBanner into the HomePage component. This way we could prevent the vertical overflowing of the page that you can see now.

**Comment by zackkrida** [comment]
@obulat No, their redirect preserves paths, so that image links or search result links out in the wild resolve correctly. You can test this with a URL like search.creativecommons.org/photos/19e2e417-a819-4761-b838-d648b92d48ce?referrer=creativecommons.org

**Review by krysal — APPROVED** [review]
+1 to Sara's suggestion about tests but glad it's working SSR.

**Comment by obulat** [comment]
Oh, thank you for explaining, @zackkrida !

File: src/middleware/middleware.js (hunk 1 above)
Comment (dhruvkb) [reply src/middleware/middleware.js:31]: This is classic half-knowledge stuff. I didn't even know there was a `process.client`!

**Review by dhruvkb — COMMENTED** [review]
I'll add an E2E test-case for this. But I've never written one before so there might be some delay due to the learning involved.

**Pull request merged**